src/hooks/useProposal/useProposal.ts

```diff
@@ -32,6 +33,7 @@ export function useProposal(proposal: ProposalTypes.Struct) {
   const accounts = useAccounts(client);
   const web3wallet = useAtomValue(web3walletAtom);
   const analytics = useAnalytics();
+  const { addAppToLastConnectionApps } = useRecentConnection();
```

Comment:
We should replace this with the `useSetAtom` here to avoid subscribing to the atom if we only need the setter function
```suggestion
  const addAppToLastConnectionApps = useSetAtom(sortedRecentConnectionApps);
```